opty/utils.py

```diff
@@ -229,6 +230,136 @@ def parse_free(free, n, q, N):
     return free_states, free_specified, free_constants
 
 
+def create_objective_function(
+    objective, state_symbols, input_symbols, unknown_symbols,
+    N, node_time_interval=1.0, integration_method="backward euler"):
+    """Creates function to evaluate the objective and objective gradient.
+
+    Parameters
+    ----------
+    objective : sympy.Expr
+        The objective function to be minimized, which is a function of the
+        states and inputs.
```

Comment:
Does this work with any kind of objective function? If not, can you say what its limits are?

Reply:
Added some clarification. Additionally, I switched to crawling the expression tree myself instead of using replace with a wildcard. This allows to raise NotImplementedErrors for all integral versions which are not supported.

Reply:
Additionally, it is also a bit more future proof and faster, as one can now easily access the limits for parsing purposes.